provisioner/aws-manager.js

```diff
@@ -70,15 +102,75 @@ AwsManager.prototype.update = function () {
         },
       ],
     }),
+    that.ec2.describeInstances({
+      Filters: [
+        {
+          Name: 'key-name',
+          Values: [that.keyPrefix + '*'],
+        },
+        {
+          Name: 'instance-state-name',
+          Values: ['shutting-down', 'terminated', 'stopping'],
+        },
+      ],
+    }),
+    that.ec2.describeSpotInstanceRequests({
+      Filters: [
+        {
+          Name: 'launch.key-name',
+          Values: [that.keyPrefix + '*'],
+        }, {
+          Name: 'state',
+          Values: ['cancelled', 'failed', 'closed', 'active'],
+        },
+      ],
```

Comment:
How about making two requests? and then filter them afterwards... That is likely much more efficient.

-----
**Unrelated detail**, do consider using `await/async`, you can do things like `var [instances, spotRequests] = await Promise.all([...` (destructive array assignments rock with `Promise.all`)
You'll be so happy, just write `AwsManager.prototype.update = async function` and it works.